### PR TITLE
#150397731 show all exercise info

### PR DESCRIPTION
**Pull request by nelmandela**
**#150397731 Show all infos for an exercise**

**What does this PR do?**

Creates a new read-only API endpoint that collects all information of an exercise such as muscles, images & category

**Description of Task to be completed?**

Have the following endpoint working;

*GET  /api/v1/exercise/view/*

**How should this be manually tested?**

* Clone the repo, cd into into it and RUN python manage.py run server ;
* Then open it on localhost & try with different exercises id using this url;
 *localhost:8000/api/v2/exercise/view/?id=exercise_id*

**What are the relevant pivotal tracker stories?**

#150397731

image

### CODE REVIEW

File: wger/exercises/api/views.py [hunk 1]
```diff
@@ -123,6 +123,53 @@ def search(request):
...
+                                    language_code=request.GET.get('language', None))
+    exercise = (Exercise.objects.filter(pk=id)
+                .filter(status=Exercise.STATUS_ACCEPTED).first())
+    json_response = {}
+
+    if exercise:
+        if exercise.main_image:
+            image_obj = exercise.main_image
+            image = image_obj.url
+            t = get_thumbnailer(image_obj.image)
```
Comment (ghost) [inline wger/exercises/api/views.py]: t?? rename the variables to be more meaningful

File: wger/exercises/api/views.py [hunk 2]
```diff
@@ -123,6 +123,53 @@ def search(request):
...
+            thumbnail = t.get_thumbnail(aliases.get('micro_cropped')).url
+        else:
+            image = None
+            thumbnail = None
+
+        muscles = [muscle.name for muscle in exercise.muscles.all()]
+
+        muscles_secondary = [muscles_secondary.name
```
Comment (ghost) [inline wger/exercises/api/views.py:150]: good use of list generators

File: wger/exercises/api/views.py [hunk 3]
```diff
@@ -123,6 +123,53 @@ def search(request):
...
+            'exercise_id': exercise.id,
+            'exercise_name': exercise.name,
+            'muscles': muscles,
+            'muscles_secondary': muscles_secondary,
+            'equipment': equipment,
+            'image': image,
+            'thumbnail': thumbnail
+        }
+
+        print(exercise.equipment.all())
```
Comment (ghost) [inline wger/exercises/api/views.py]: remove print statements unless needed

**Review by ghost — CHANGES_REQUESTED** [review]
Good job, I only have a few minor fixes and we can merge.

File: wger/exercises/api/views.py (hunk 1 above)
Comment (nelmandela) [reply wger/exercises/api/views.py]: thanks for the observation and suggestions.lets work on that then we update

**Review by ghost — APPROVED** [review]

**Pull request merged**